virttest/qemu_vm.py

```diff
@@ -1632,83 +1535,47 @@ def sort_key(dev):
                 devices.insert(pvpanic_dev)
 
         # Add serial console redirection
-        for serial in params.objects("serials"):
+        serials = params.objects('serials')
+        if serials:
+            self.serial_session_device = serials[0]
+            host = params.get('chardev_host', '127.0.0.1')
+            free_ports = utils_misc.find_free_ports(
+                5000, 6000, len(serials), host)
+            reg_count = 0
+        self.virtio_ports = []
+        for index, serial in enumerate(serials):
             serial_filename = vm.get_serial_console_filename(serial)
```

Comment:
Since  isa-serial  and  virtio serial use same method to get `serial_filename` now. I wonder if we need to update  function `get_virtio_port_filename` in `virt_vm.py`.  Since it use  `virtio_port-%s-%s" % (port_name, self.instance)`. 

It may do not affect our test, as in qemu_vm  `get_virtio_port_filename`  is ok.

Reply:
@vivianQizhu   correct my comment above,   qemu_vm use `get_virtio_port_filenames`  not `get_virtio_port_filename`.   only one `get_virtio_port_filename` in virt_vm.  It is not correct now.

Reply:
@yanan-fu I see the two impact ones: https://github.com/autotest/tp-qemu/blob/master/qemu/tests/qemu_guest_agent.py#L199 and https://github.com/autotest/tp-qemu/blob/master/qemu/tests/qemu_guest_agent_hotplug.py#L32
What do you think if I replace them with `get_serial_console_filename`? Cause now we actually totally abandon the usage of `get_virtio_port_filename` in qemu_vm context.

Reply:
@yanan-fu And for qemu_vm.py:
```
diff --git a/virttest/qemu_vm.py b/virttest/qemu_vm.py
index 9fc849c9..b0c51f62 100644
--- a/virttest/qemu_vm.py
+++ b/virttest/qemu_vm.py
@@ -3018,7 +3018,6 @@ class VM(virt_vm.BaseVM):
         # Generate the tmp file which should be deleted.
         file_list = [self.get_testlog_filename()]
         file_list += qemu_monitor.get_monitor_filenames(self)
-        file_list += self.get_virtio_port_filenames()
         file_list += self.get_serial_console_filenames()
         file_list += list(self.logs.values())
```

Reply:
@vivianQizhu  it is ok for me to use `get_serial_console_filename` in case level. as when generate qemu command line, we use same method to generate filename.   
And  remove `get_virtio_port_filename` and `get_virtio_file_names` directly ?   i have checked tp-libvirt, no use.

Reply:
@yanan-fu Even it is not used by tp-libvirt now, I don't really want to change the base class, I only care how qemu_vm works.

Reply:
At least,  the one in qemu_vm need be deleted, `get_virtio_port_filenames`. 

For virt_vm, i prefer to delete the two function.  As my understanding, only qemu care it but do not need it.  

@xutian  do you have any idea about this issue ?

Reply:
@yanan-fu Okay, all deleted, hope has no side effect.